moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -91,8 +91,17 @@ double RobotTrajectory::getDuration() const
 
 double RobotTrajectory::getAverageSegmentDuration() const
 {
-  if (duration_from_previous_.empty())
-    return 0.0;
+  // If the initial segment has a timstep of 0, exclude it grom teh average calculation
+  if (duration_from_previous_[0] == 0)
+  {
+    if (duration_from_previous_.size() <= 1)
+    {
+      RCLCPP_WARN(rclcpp::get_logger("RobotTrajectory"), "Too few waypoints to calculate a duration. Returning 0.");
+      return 0.0;
+    }
```

Comment:
I think this error checking should be moved outside of `if (duration_from_previous_[0] == 0)` since it applies to either case

Reply:
I was thinking about this: if you have one waypoint with a non-zero `duration_from_previous`, wouldn't moving this check return zero rather than the value?

Reply:
OK, good point. Well, I think the `else` case here needs a bit of error checking too like it originally had. Will make a suggestion.